pytket_qir/qir.py

```diff
@@ -103,17 +103,15 @@ class QirParser:
     def __init__(
         self,
         file_path: str,
-        gateset: Optional[CustomGateSet],
-        wasm_handler: Optional[WasmFileHandler],
-        wasm_int_type: Optional[types.Int] = types.Int(32),
-        qir_int_type: Optional[types.Int] = types.Int(64),
+        gateset: CustomGateSet = None,
+        wasm_handler: WasmFileHandler = None,
```

Comment:
Since `None` isn't a valid value for `CustomGateSet` or `WasmFileHandler`, these type annotations don't make sense. If we want `None` to be a default then they should be `Optional[...]`.

Reply:
There is a thing here, for instance, I tried to default `gateset` to `PYQIR_GATES` but the default value didn't seem to be propagated further the initialiser. I.e. `gateset` was `NoneType` even if defaulted to `PYQIR_GATES`. Unless I did something egregiously wrong, my understanding is that (at least in that case) handling a default value works if it is set to `None` ? 

Ah, OK I see what you mean: `None` isn't a valid `CustomGateSet` type. Hm, I get it better but nonetheless, I couldn't default it to `PYQIR_GATES` as I attempted above. Other point, why `mypy` doesn't pick up these non-sensical type situations ? 

Let me dig in that a little more.

Reply:
Done. This is still weird: I can't default `gateset` tp `PYQIR_GATE` as I would like to so making it `Optional`.